corehq/motech/repeaters/tests/test_models.py

```diff
@@ -627,6 +627,55 @@ def test_count_overdue(self):
         overdue = SQLRepeatRecord.objects.count_overdue()
         self.assertEqual(overdue, 3)
 
+    iter_partition = SQLRepeatRecord.objects.iter_partition
+
+    def test_one_partition(self):
+        iter_partition = type(self).iter_partition
+        all_ids = self.make_records(5)
+        start = datetime.utcnow()
+        ids = {r.record_id for r in iter_partition(start, 0, 1)}
+        self.assertEqual(ids, all_ids)
+
+    def test_four_partitions(self):
+        iter_partition = type(self).iter_partition
+        all_ids = self.make_records(16)
+        start = datetime.utcnow()
+        ids0 = {r.record_id for r in iter_partition(start, 0, 4)}
+        ids1 = {r.record_id for r in iter_partition(start, 1, 4)}
+        ids2 = {r.record_id for r in iter_partition(start, 2, 4)}
+        ids3 = {r.record_id for r in iter_partition(start, 3, 4)}
+
+        self.assertEqual(ids0 | ids1 | ids2 | ids3, all_ids)
```

Comment:
Rather than all of the asserts below, what do you think about returning lists from the `iter_partition` calls above, and then just ensuring that adding up all of the lists is equivalent to `all_ids`. Something like:

```suggestion
        ids0 = [r.record_id for r in iter_partition(start, 0, 4)]
        ids1 = [r.record_id for r in iter_partition(start, 1, 4)]
        ids2 = [r.record_id for r in iter_partition(start, 2, 4)]
        ids3 = [r.record_id for r in iter_partition(start, 3, 4)]

        self.assertCountEqual(ids0 + ids1 + ids2 + ids3, all_ids)
```
And since these records are being saved to the DB, we can safely assume that `all_ids` is a unique list.

Reply:
ef72e9d483c1266d31208773a71a36ec96f693b6